### DISCUSSION TITLE
关于修改目录中“章”的缩进

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#168] [Q&A]
您好，
我修改了**章**的显示为：
![image](https://user-images.githubusercontent.com/45897471/187832754-4bc71ed1-b77f-4779-8b69-0ed5cb2d7f53.png)
其中涉及的代码为：
```
\tl_if_eq:NnF \g__whu_option_type_tl { opening }
  {
    \ctexset
      {
        chapter        =
          {
            format     = { \centering \heiti \zihao { -2 } },
            number     = { 第\,\chinese{chapter} \,章},%原来为：\arabic { chapter }           
            name       = {,    },
            beforeskip = { 0.8 \c__whu_baseline_skip       },
            afterskip  = { 0.5 \c__whu_baseline_skip       },
            hang=true %此行是增加的
          }  % 各章标题 黑体小 2 号
      }
  }
```
但是目录中的显示为：
![image](https://user-images.githubusercontent.com/45897471/187832874-289966f8-52dd-4932-a093-74574aa63c20.png)
我应该怎么做才可以使得目录中的缩进在**章**的内容处对齐，目前是在“一”这个位置对齐。
另一个问题是，当我对**章**进行引用时（`\ref{sec:   }`），比如引用后显示为：第一章，但是这个**第一章**文字间距过大，因为在修改的代码中增加了`\,`。在不取消`\,`的情况下，我可以使得对**章**引用时自动删除`\,`这个间距吗？
比如：
```
\chapter{这是第一章标题的测试这是第一章标题的测试这是第一章标题的测试这是第一章标题的测试这是第一章标题的测试这是第一章标题的测试这是第一章标题的测试这是第一章标题的测试}\label{sec:1}
我们\ref{sec:1}我们
```
显示效果为：
![image](https://user-images.githubusercontent.com/45897471/187834175-56b9b194-cf17-4618-98e4-4fcd3335888b.png)
很明显**一**与**第**、**章**的间距过大。使用的是武大的模板，其中**cls**文件已经修改，我把**cls**文件放在附件中，
[whu-thesis.zip](https://github.com/whutug/whu-thesis/files/9467062/whu-thesis.zip)，请把代码复制到模板中的**chapter1.tex**。

谢谢！

**Comment by xkwxdyy** [comment]
> 在不取消\,的情况下，我可以使得对章引用时自动删除\,这个间距吗？

那你加的意义何在

**Reply by xkwxdyy** [reply]
用阿拉伯数字不好吗？
![image](https://user-images.githubusercontent.com/87134063/187833664-2c7ea9ec-525c-48db-bea8-24800bbf1618.png)

**Reply by earthmoon** [reply]
> > 在不取消,的情况下，我可以使得对章引用时自动删除,这个间距吗？
> 
> 那你加的意义何在

就是在文中**章**的标题中好看一点，可是在正文中要是文字（比如：第一章）间距过大则会不协调。

**Reply by earthmoon** [reply]
> 用阿拉伯数字不好吗？ ![image](https://user-images.githubusercontent.com/87134063/187833664-2c7ea9ec-525c-48db-bea8-24800bbf1618.png)

我看课题组之前采用的都是**第几章**这样的**章**的标题

**Reply by xkwxdyy** [reply]
你的引用之类的，没有完整的 MWE。

**Reply by xkwxdyy** [reply]
> 我看课题组之前采用的都是第几章这样的章的标题

博士论文没有对目录的样式设置要求。“课题组之前”完全是由于之前用的模版不同，那你为何不直接用他们的，还省的你自己操心格式问题

**Reply by earthmoon** [reply]
> > 我看课题组之前采用的都是第几章这样的章的标题
> 
> 博士论文没有对目录的样式设置要求。“课题组之前”完全是由于之前用的模版不同，那你为何不直接用他们的，还省的你自己操心格式问题

我更新了问题，现在有截图和部分代码。那是word的，大文档很卡，而且越到后面用的越累，因为改起来工作量会很大。

**Reply by xkwxdyy** [reply]
你提了这么多问还是不知道什么叫 MWE 吗
![image](https://user-images.githubusercontent.com/87134063/187836267-e8e921d9-9413-4852-9ec8-35ab1ad989c6.png)

**Reply by earthmoon** [reply]
我修改了问题，现在问题中提供了**cls**文件，其他文字只需要复制以下就可以了。
